### PR TITLE
Fix #15706: navbar-text margin right in top fixed fluid container

### LINKED ISSUE #15706: navbar-text floating to right in top fixed fluid container navbar issue in 3.3.2
I found a problem when trying to add .navbar-text .navbar-right in top-fixed-navbar with fluid container using 3.3.2. The text doesn't have negative right margin so that it too close to screen border. It work just fine if using nav list. [Here example of the problem](http://www.bootply.com/mvsni9t5bI). It has negative right margin when using 3.1.1.

### PR DESCRIPTION
**Pull request by kkirsche**
Fixes #15706 navbar-text margin right in top fixed fluid container

Build failed due to Travis Error. Please trigger rebuild:

``` shell
Loading "autoprefixer.js" tasks...ERROR
>> Error: Cannot find module 'ansi-regex'
Warning: Task "autoprefixer:core" not found. Use --force to continue.
```

### CODE REVIEW

**Comment by kkirsche** [comment]
Build failed due to Travis Error. Please trigger rebuild:

```shell
Loading "autoprefixer.js" tasks...ERROR
>> Error: Cannot find module 'ansi-regex'
Warning: Task "autoprefixer:core" not found. Use --force to continue.
```

**Comment by kkirsche** [comment]
Much better, thanks

**Comment by mdo** [comment]
I might be fine punting on this for v3. It's not mission critical and changing the navbar is just the worst.

**Comment by kkirsche** [comment]
I can understand that. I think it's a helpful thing to fix so that `.container` and `.container-fluid` work as expected but I can understand wanted to punt it and just use the complete revision in V4.

File: less/navbar.less [hunk 1]
```diff
@@ -118,6 +118,12 @@
   }
 }
 
+.container-fluid {
+  > .navbar-text.navbar-right:last-child{
```
Comment (XhmikosR) [inline less/navbar.less]: \<nitpicking\> Missing space before the opening bracket here. \</nitpicking\>

**Comment by kkirsche** [comment]
Good catch. Fixed!

**Comment by cvrebert** [comment]
@mdo So what's your final decision on this?

**Comment by mdo** [comment]
We don't override anything here the same components in `.container` though, do we? It feels weird to be doing it here.

**Comment by kkirsche** [comment]
Wanted to follow up about this and see if there were any adjustments that were necessary.

Thank you for your time and assistance guys

**Comment by kkirsche** [comment]
Wanted to follow up on this and see what everyone thought. Thanks for the feedback everyone!

**Comment by kkirsche** [comment]
Wanted to follow up on this and see what everyone thought. Thanks for the feedback everyone!

**Comment by cvrebert** [comment]
@mdo I'm not sure I understand your question.

**Comment by kkirsche** [comment]
Wanted to follow up on this and check if it was missing anything.

**Comment by smisch** [comment]
Yes there is some issue left with this commit. Please address https://github.com/twbs/bootstrap/issues/15706#issuecomment-77133738

**Comment by mdo** [comment]
Yeah, @smisch raises a good point—the containers aren't the immediate parents of those elements. The collapse container is required, too.

**Comment by mdo** [comment]
I'm going to go ahead and punt on this. v3's navbars have seen enough flipping around of our margins and more for their layout.

**Pull request closed without merging**